include/fiction/algorithms/simulation/sidb/operational_domain.hpp

```diff
@@ -1054,9 +991,9 @@ class operational_domain_impl
      */
     [[nodiscard]] inline std::optional<operational_status> has_already_been_sampled(const step_point& sp) const noexcept
     {
```

Comment:
The multiple distinctions between different operational domain types inside the function feel kinda clunky, no? I think this can be improved by providing dedicated implementations of `has_already_been_sampled` for each operational domain type supported via `std::enable_if`.

Reply:
Good idea! I am not sure if I understand this correctly: I would just add the corresponding function as a private function to the corresponding operational domain, right?

Reply:
I put it inside ``sidb_simulation_domain`` as a member function.